src/promptflow/promptflow/_sdk/_serving/flow_invoker.py

```diff
@@ -201,3 +201,38 @@ def _dump_invoke_result(self, invoke_result):
                 invoke_result.output, base_dir=self._dump_to, sub_dir=Path(".promptflow/output")
             )
             dump_flow_result(flow_folder=self._dump_to, flow_result=invoke_result, prefix=self._dump_file_prefix)
+
+
+class AsyncFlowInvoker(FlowInvoker):
+    async def _invoke_async(self, data: dict, run_id=None, disable_input_output_logging=False):
+        log_data = "<REDACTED>" if disable_input_output_logging else data
+        self.logger.info(f"Validating flow input with data {log_data!r}")
+        validate_request_data(self.flow, data)
+        self.logger.info(f"Execute flow with data {log_data!r}")
+        # Pass index 0 as extension require for dumped result.
+        # TODO: Remove this index after extension remove this requirement.
+        result = await self.executor.exec_line_async(
+            data, index=0, run_id=run_id, allow_generator_output=self.streaming()
+        )
+        if LINE_NUMBER_KEY in result.output:
```

Comment:
Will single line run add line number in results?

Reply:
let me double check and extract this logic in base class.